packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -206,10 +207,27 @@ Future<XcodeBuildResult> buildXcodeProject({
     }
   }
 
-  if (buildForDevice) {
-    buildCommands.addAll(<String>['-sdk', 'iphoneos']);
+  // Check if the project contains a watchOS companion app.
+  final bool hasWatchCompanion = await app.project.containsWatchCompanion(projectInfo.targets);
+  if (hasWatchCompanion) {
+    // The -sdk argument has to be omitted if a watchOS companion app exists.
+    // Otherwise the build will fail as WatchKit dependencies cannot be build using the iOS SDK.
+    globals.printStatus('Watch companion app found. Adjusting build settings.');
+    if (! buildForDevice && deviceID == '') {
+      globals.printError('No simulator device ID has been set.');
+      globals.printError('A device ID is required to build a flutter a flutter app with a companion watch app.');
```

Comment:
"build a flutter a flutter app" -> "build an app with a companion watch"
Can you mention that the ID should be passed in with the `--devices` flag?

Reply:
done